src/claimtrie.cpp

```diff
@@ -1634,156 +1096,107 @@ bool CClaimTrieCache::addClaimToQueues(const std::string& name, CClaimValue& cla
     return true;
 }
 
-bool CClaimTrieCache::removeClaimFromQueue(const std::string& name, const COutPoint& outPoint, CClaimValue& claim) const
+bool CClaimTrieCache::removeClaimFromQueue(const std::string& name, const COutPoint& outPoint, CClaimValue& claim)
 {
-    queueNameType::iterator itQueueNameRow = getQueueCacheNameRow(name, false);
-    if (itQueueNameRow == claimQueueNameCache.end())
-    {
-        return false;
-    }
+    queueNameType::iterator itQueueNameRow = getQueueCacheRow(name, claimQueueNameCache, false);
+    if (itQueueNameRow == claimQueueNameCache.end()) return false;
     queueNameRowType::iterator itQueueName;
     for (itQueueName = itQueueNameRow->second.begin(); itQueueName != itQueueNameRow->second.end(); ++itQueueName)
     {
-        if (itQueueName->outPoint == outPoint)
-        {
-            break;
-        }
-    }
-    if (itQueueName == itQueueNameRow->second.end())
-    {
-        return false;
+        if (itQueueName->outPoint == outPoint) break;
     }
-    claimQueueType::iterator itQueueRow = getQueueCacheRow(itQueueName->nHeight, false);
-    if (itQueueRow != claimQueueCache.end())
-    {
+    if (itQueueName == itQueueNameRow->second.end()) return false;
+    claimQueueType::iterator itQueueRow = getQueueCacheRow(itQueueName->nHeight, claimQueueCache, false);
+    if (itQueueRow != claimQueueCache.end()) {
         claimQueueRowType::iterator itQueue;
-        for (itQueue = itQueueRow->second.begin(); itQueue != itQueueRow->second.end(); ++itQueue)
-        {
-            if (name == itQueue->first && itQueue->second.outPoint == outPoint)
-            {
-                break;
+        for (itQueue = itQueueRow->second.begin(); itQueue != itQueueRow->second.end(); ++itQueue) {
+            if (name == itQueue->first && itQueue->second.outPoint == outPoint) {
+                std::swap(claim, itQueue->second);
+                itQueueNameRow->second.erase(itQueueName);
+                itQueueRow->second.erase(itQueue);
+                base->removeFromClaimIndex(claim);
+                return true;
             }
         }
-        if (itQueue != itQueueRow->second.end())
-        {
-            std::swap(claim, itQueue->second);
-            itQueueNameRow->second.erase(itQueueName);
-            itQueueRow->second.erase(itQueue);
-
-            base->removeFromClaimIndex(claim);
-            return true;
-        }
     }
     LogPrintf("%s: An inconsistency was found in the claim queue. Please report this to the developers:\nFound in named queue but not in height queue: name: %s, txid: %s, nOut: %d, nValidAtHeight: %d, current height: %d\n", __func__, name, outPoint.hash.GetHex(), outPoint.n, itQueueName->nHeight, nCurrentHeight);
     return false;
 }
 
-bool CClaimTrieCache::undoAddClaim(const std::string& name, const COutPoint& outPoint, int nHeight) const
+bool CClaimTrieCache::undoAddClaim(const std::string& name, const COutPoint& outPoint, int nHeight)
```

Comment:
Why are we removing const from all of the methods?  Are we also removing mutable members?

Reply:
Yes.

Reply:
I like the change. Saying that a removal method is `const` is misleading. The only place I would expect to see the `mutable` keyword is in an immutable object for the caching of expensive-to-compute fields.